monkey/infection_monkey/i_puppet/i_puppet.py

```diff
@@ -59,11 +60,16 @@ def run_credential_collector(self, name: str, options: Dict) -> Sequence[Credent
         """
 
     @abc.abstractmethod
-    def run_pba(self, name: str, options: Dict) -> PostBreachData:
+    def run_pba(
+        self, name: str, telemetry_messenger: ITelemetryMessenger, options: Dict
+    ) -> PostBreachData:
```

Comment:
Can we pass the telemetry messenger to each PBA when it is constructed and loaded into the puppet, rather than when it is run? This would let us give different messengers to different PBAs.

Reply:
I'll fix it